src/dattobd.c

```diff
@@ -483,7 +508,7 @@ static inline void dattobd_inode_unlock(struct inode *inode){
 #define tracer_should_trace_bio(dev, bio) (bio_data_dir(bio) && !(bio->bi_rw & REQ_DISCARD) && bio_size(bio) && !tracer_read_fail_state(dev) && tracer_sector_matches_bio(dev, bio))
 
 //macros for verifying file
-#define file_is_on_bdev(file, bdev) ((dattobd_get_mnt(file))->mnt_sb == (bdev)->bd_super)
+#define file_is_on_bdev(file, bdev) ((dattobd_get_mnt(file))->mnt_sb == dattobd_get_super_free(bdev))
```

Comment:
change `file_is_on_bdev()` to a static inline function that does the correct reference handling. Move the declaration down to just above where `file_is_on_bdev()` is called.